app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,250 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Matrix;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import com.camerakit.CameraKitView;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+
+/**
+ * The Activity useful for making photos
+ */
+public class CameraActivity extends AppCompatActivity {
+
+    private CameraKitView cameraKitView;
+    private static String orientationResult;
+
+    /**
+     * onCreate method of the Android Activity Lifecycle
+     * @param savedInstanceState The Bundle of the last instance state saved
+     * @author Romanello Stefano
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_camera);
+
+        cameraKitView = findViewById(R.id.cameraKitView);
+
+        //Load sensor for understand the orientation of the phone
+        SensorManager sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+        sensorManager.registerListener(new SensorEventListener() {
+            public int mOrientationDeg; //last rotation in degrees
+            private static final int _DATA_X = 0;
+            private static final int _DATA_Y = 1;
+            private static final int _DATA_Z = 2;
+            private int ORIENTATION_UNKNOWN = -1;
+
+            @Override
+            public void onSensorChanged(SensorEvent event)
+            {
+                float[] values = event.values;
+                int orientation = ORIENTATION_UNKNOWN;
+                float X = -values[_DATA_X];
+                float Y = -values[_DATA_Y];
+                float Z = -values[_DATA_Z];
+                float magnitude = X*X + Y*Y;
+                // Don't trust the angle if the magnitude is small compared to the y value
+                if (magnitude * 4 >= Z*Z) {
+                    float OneEightyOverPi = 57.29577957855f;
```

Comment:
Could be better to add final on OneEightyOverPi or define it at the beginning of the method

Reply:
i add final or private to the variable to avoid other method to modify it